multicluster/service-mirror/cluster_watcher.go

```diff
@@ -655,6 +708,71 @@ func (rcsw *RemoteClusterServiceWatcher) Start(ctx context.Context) error {
 			},
 		},
 	)
+	if rcsw.headlessServicesEnabled {
+		rcsw.remoteAPIClient.Endpoint().Informer().AddEventHandler(
+			cache.ResourceEventHandlerFuncs{
+				AddFunc: func(obj interface{}) {
```

Comment:
I extracted some of the logic out into a separate function `shouldProcessEndpoints()`. I'm also in favour of adding a "filtered" informer; I think @alpeb and I talked about it as a next step but if you think it might be worthwhile to do as part of this change, I'm all for it.